k8s-plugin-keel/src/main/kotlin/com/amazon/spinnaker/keel/k8s/verificationEvaluator/K8sJobEvaluator.kt

```diff
@@ -101,7 +99,24 @@ class K8sJobEvaluator(
         }
     }
 
+    // need to randomize name otherwise subsequent jobs may fail with the same name in the same namespace
+    // orca, clouddriver, or keel cleans finished jobs.
     private fun generateOrcaK8sJob(application: String, verification: K8sJobVerification): OrcaJob {
+        verification.manifest[API_VERSION] = VERIFICATION_K8S_JOB_API_V1
+        verification.manifest[KIND] = VERIFICATION_K8S_JOB_KIND
+
+        val uuid = UUID.randomUUID().toString()
+        val metadata = if (verification.manifest.containsKey("metadata")) {
+            @Suppress("UNCHECKED_CAST")
+            verification.manifest["metadata"] as MutableMap<String, Any>
+        } else {
+            mutableMapOf()
+        }
+
+        metadata[NAME] = verification.jobNamePrefix?.let {
```

Comment:
This can easily result in a job name being > 63 characters which causes the job to fail.

Reply:
Ah good point. I thought the limit was 263 but I was wrong.